### DISCUSSION TITLE
ReferenceError: Property 'React' doesn't exist; located at: Web3Modal

### DISCUSSION THREAD

**Discussion by yashwanthmuddana** [WalletConnect/walletconnect-monorepo#2701] [Developer Support]
Trying to migrate from v1 to v2 in react-native
- version 
 ```
  "react": "17.0.2",
   "react-native": "0.67.5",
```

- imported 
```
    import '@walletconnect/react-native-compat';
    import {Web3Modal} from '@web3modal/react-native';
```
-  usage
    `<Web3Modal
          projectId={projectId}
          providerMetadata={providerMetadata}
          sessionParams={sessionParams}
        />`

- Getting ReferenceError: Property 'React' doesn't exist,
- React is Declared already in n /node_modules/@web3modal/react-native/lib/commonjs/components/Web3Modal.js
- every thing is working fine in v1. could i get any help on this

**Comment by finessevanes** [comment]
Could you give us a minimal reproducible example?

**Reply by yashwanthmuddana** [reply]
yes, we can reproduce this error with this version of react and react-native

`"react": "17.0.2",
  "react-native": "0.67.5",
`

**Reply by yashwanthmuddana** [reply]
i have created a new app with version 
`"react": "18.2.0",
  "react-native": "0.71.11",`
just for testing, and it is working.  i guess there is some problem with the react versions.. could you let me know if we have any work arounds for `"react": "17.0.2", "react-native": "0.67.5"`
